pennylane/tape/tape.py

```diff
@@ -930,6 +936,268 @@ def set_parameters(self, params, trainable_only=True):
+
+            def func(a):
+                qml.RY(a[0], wires=0)
+                qml.RX(a[1], wires=0)
+                qml.RY(a[2], wires=0)
+
+            with qml.tape.QuantumTape() as tape:
+                func(a)
+                qml.state()
+
+            # 1. Execute the tape
+            tape_execute_res = qml.execute(
+                [tape],
+                dev,
+                cache=None,
+                interface="autograd",
+                gradient_fn=param_shift
+            )
+
+            # Get the result of the first tape execution
+            tape_execute_res = tape_execute_res[0]
+
+        .. code-block:: pycon
+
+            >>> print(tape_execute_res.shape)
+            (1, 4)
+
+        There is a difference between the shape of results obtained by:
+
+        1. Using the low-level ``qml.execute`` pipeline for executing a ``QuantumTape``;
+        2. Evaluating a QNode.
+
+        The shape returned by this method corresponds to case 1. QNode
+        evaluation may include an additional squeezing step.
+
+        For example, for `qml.state` the result in each case would be (see
+        example below):
+
+        1. ``(1,4)``;
+        2. ``(4,)``
+
+        .. code-block:: python
+
+            # 2. Evaluate the QNode
+            @qml.qnode(dev,interface='autograd')
+            def circuit(a):
+                func(a)
+                return qml.state()
+
+            res_qnode = circuit(a)
+
+        .. code-block:: pycon
+
+            >>> print(tape_execute_res.shape, res_qnode.shape)
+            (1, 4) (4,)
+
+        Args:
+            device (.Device): the device that will be used for the tape execution
+
+        Raises:
+            TapeError: raised for unsupported cases for
+                example when the tape contains heterogeneous measurements
+
+        Returns:
+            Union[tuple[int], list[tuple[int]]]: the output shape(s) of the
+            tape result
+        """
+        output_shape = tuple()
+
+        if len(self._measurements) == 1:
+            output_shape = self._single_measurement_shape(self._measurements[0], device)
+        else:
+            num_measurements = len(set(meas.return_type for meas in self._measurements))
+            if num_measurements == 1:
+                output_shape = self._multi_homogenous_measurement_shape(self._measurements, device)
+            else:
+                raise TapeError(
+                    "Getting the output shape of a tape that contains multiple types of measurements is unsupported."
+                )
+        return output_shape
+
+    def numeric_type(self):
+        """Returns the expected numeric type of the tape result by inspecting
+        its measurements.
+
+        Raises:
+            TapeError: raised for unsupported cases for
+                example when the tape contains heterogeneous measurements
+
+        Returns:
+            type: the numeric type corresponding to the result type of the
+            tape
```

Comment:
worth having an example here?

Reply:
Added in https://github.com/PennyLaneAI/pennylane/pull/2044/commits/7e4f99e962dab0459125f33a3efea1bb713d3f0a